department-of-veterans-affairs/vets-website
# Add military base checkbox

## Summary

On the Step 1 Applicant Information page, adds a new question for "I receive mail outside of the United States on a U.S. military base".

Per discussion with @mayacoforma, the custom warning label for postal code hasn't been added, instead relying on the existing postal code validation. 

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Select 'Start your application without signing in'
3. Verify the page 'Mailing address' is in the correct location, per the designs, and includes only one 'Mailing address' title
4. Verify the page 'Mailing address' has a checkbox for "I receive mail outside of the United States on a U.S. military base" and has the dropdown for "Learn more about military base addresses"
5. Verify that when the checkbox for "I receive mail outside of the United States on a U.S. military base" is selected, the postal code must be a five-digit number.

## Related issue(s)

- [GitHub Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/70329)
- [Epic in GitHub](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)


## Testing done

## Screenshots

| Before | After | Validation |
| ------- | ------ | --------- |
|  <img width="431" alt="mailing" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/1cc4e5cc-4380-4ea1-bce4-55fe69b2d897"> | <img width="454" alt="After" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/edf3b34d-5a70-4a04-88bd-8eded40c260d"> | <img width="458" alt="Screenshot 2023-11-29 at 3 13 12 PM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/867b1c1c-b01e-4f2e-95c5-fc329b953ccf"> |

## What areas of the site does it impact?

Benefits Pensions

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication
